src/object/class_object.c

```diff
@@ -7986,15 +8013,6 @@ classobj_check_index_compatibility (SM_CLASS_CONSTRAINT * constraints, const DB_
 	  ret = SM_CREATE_NEW_INDEX;
 	  return ret;
 	}
-      else if (SM_IS_SHARE_WITH_FOREIGN_KEY (existing_con->type))
-	{
-	  ret = SM_SHARE_INDEX;
-	  if (existing_con->filter_predicate != NULL || existing_con->func_index_info != NULL)
-	    {
-	      ret = SM_CREATE_NEW_INDEX;
-	    }
-	  return ret;
-	}
```

Comment:
Is it safe to remove this routine? `SM_IS_SHARE_WITH_FOREIGN_KEY` checks not only `SM_CONSTRAINT_FOREIGN_KEY` but `SM_CONSTRAINT_INDEX`. `SM_CONSTRAINT_INDEX` is added for partition constraints and I found some SQL regressions related to partitions.

Reply:
In this routine, existing_con->type is always SM_CONSTRAINT_FOREIGN_KEY because the classobj_find_constraint_by_attrs is changed to return constraint SM_CONSTRAINT_FOREIGN_KEY for DB_CONSTRAINT_FOREIGN_KEY.

Reply:
Ok, I confused.

Reply:
I'm gonna change the routine as below:
`else if (existing_con->type == SM_CONSTRAINT_INDEX)
  {
    AS_IS
  }`

Reply:
It looks better to modify of **classobj_find_constraint_by_attrs** to return NULL when it can't find existing foreign key.
And this classobj_check_index_exist checks just only existing index, so I think that it's good to return SM_CREATE_INDEX at exising_con = NULL in foreign key.

Reply:
the routine is changed as below:
     else if (existing_con->type == SM_CONSTRAINT_INDEX)
	{
            return NEW or SHARE as compatibility specification.
        }